### ISSUE TITLE
DSA issue in the presence of return structures

### ISSUE THREAD

**Issue by shaobo-he** [smackers/smack#507]
DSA seems unable to handle code with return values that have structure types. For example,

```C
#include "smack.h"

int* A;

typedef struct {
  int* a;
  long b;
} S;

S foo() {
  S x = {A, 2L};
  assert(1);
  return x;
}

int main() {
  int a = 1;
  A = &a;
  S y = foo();
  assert(*(y.a) == 1);
  return 0;
}
```

When SMACK is invoked with flag ` --clang-options=-O1`, an error trace is returned. However, if we add flag `--no-memory-splitting`, no errors are reported. I generated the Boogie program and found that loading `y.a` accesses a map that is never updated.

**Comment by zvonimir** [comment]
DSA has an LLVM pass that tries to get rid of such constructs before the algorithm starts. This pass is kind of brittle I think, and so I can easily see that it does not always work. Given that they have this passs in the first place, I can easily see that DSA does not work properly if returns of structures make their way to the main DSA algorithm. Just my hunch.

**Comment by shaobo-he** [comment]
Fixed via #550
